planning/MartenV4.md

```diff
@@ -0,0 +1,114 @@
+# Marten v4 Ideas
+
+## Pulling it Off
+
+We've got the typical problems of needing to address incoming pull requests and bug issues in master while probably needing to have a long lived branch for *v4*. 
+
+Here are some thoughts:
+
+* First do a bug sweep 3.12 release to address as many of the tactical problems as we can before branching to v4?
+* Start with the unit testing improvements as a way to speed up the build before we dive into much more of this? 
+* Possibly do a series of v4, then v5 releases to do this in smaller chunks? We've mostly said do the event store as v4, then Linq improvements as v5
```

Comment:
Sure we don't need to deliver everything at once, but in v4.0.0 we should squeeze all the breaking changes to reduce the confusion.

Reply:
I agree with this. I know we talked about doing v4 for events, then v5 for other things, but Id rather just do it in one larger release if there's going to be breaking changes.